helix-cli/src/metrics_sender.rs

```diff
@@ -0,0 +1,413 @@
+        match message {
+            MetricsMessage::Event(event) => {
+                if let Some(ref mut writer) = log_writer
+                    && let Err(e) = write_event_to_log(writer, &event)
+                {
+                    eprintln!("Failed to write metrics event: {e}");
+                }
+            }
+            MetricsMessage::Shutdown => {
+                break;
+            }
+        }
+    }
+
+    if let Some(mut writer) = log_writer {
+        let _ = writer.flush();
+    }
+
+    Ok(())
+}
+
+pub(crate) fn load_metrics_config() -> Result<MetricsConfig> {
+    let config_path = get_metrics_config_path()?;
+
+    if !config_path.exists() {
+        return Ok(MetricsConfig::default());
+    }
+
+    let content = fs::read_to_string(&config_path)?;
+    let config = toml::from_str(&content)?;
+    Ok(config)
+}
+
+pub(crate) fn save_metrics_config(config: &MetricsConfig) -> Result<()> {
+    let config_path = get_metrics_config_path()?;
+    let content = toml::to_string_pretty(config)?;
+    fs::write(&config_path, content)?;
+    Ok(())
+}
+
+pub(crate) fn get_metrics_config_path() -> Result<PathBuf> {
+    let home = home_dir().ok_or_else(|| eyre!("Cannot find home directory"))?;
+    let helix_dir = home.join(".helix");
+    fs::create_dir_all(&helix_dir)?;
+    Ok(helix_dir.join("metrics.toml"))
+}
+
+fn get_metrics_dir() -> Result<PathBuf> {
+    let home = home_dir().ok_or_else(|| eyre!("Cannot find home directory"))?;
+    let metrics_dir = home.join(".helix").join("metrics");
+    fs::create_dir_all(&metrics_dir)?;
+    Ok(metrics_dir)
+}
+
+fn create_log_writer(path: &PathBuf) -> Result<BufWriter<File>> {
+    let file = OpenOptions::new().create(true).append(true).open(path)?;
+    Ok(BufWriter::new(file))
+}
+
+fn write_event_to_log<W: Write>(writer: &mut W, event: &RawEvent<EventData>) -> Result<()> {
+    let json = serde_json::to_string(event)?;
+    writeln!(writer, "{json}")?;
+    writer.flush()?;
+    Ok(())
+}
+
+async fn upload_previous_logs() -> Result<()> {
+    let metrics_dir = get_metrics_dir()?;
+    let client = Client::new();
+    let today = Local::now().date_naive();
+
+    let entries = fs::read_dir(&metrics_dir)?;
+
+    for entry in entries {
+        let entry = entry?;
+        let path = entry.path();
+
+        if let Some(file_name) = path.file_name().and_then(|n| n.to_str())
+            && let Some(date_str) = file_name.strip_suffix(".json")
+            && let Ok(file_date) = NaiveDate::parse_from_str(date_str, "%Y-%m-%d")
+            && file_date < today
+            && upload_log_file(&client, &path).await.is_ok()
+        {
+            let _ = fs::remove_file(&path);
+        }
+    }
+
+    Ok(())
+}
+
+async fn upload_log_file(client: &Client, path: &PathBuf) -> Result<()> {
+    let content = fs::read_to_string(path)?;
+
+    if content.trim().is_empty() {
+        return Ok(());
+    }
+
+    let response = client
+        .post("https://logs-dummy.helix-db.com/")
```

Comment:
There's still a dummy endpoint, how logs are uploaded are still TBD, this is one request per file but unsure if current endpoint would accept a jsonl body